crypto/signatures.go

```diff
@@ -69,8 +72,47 @@ func VerifyHash(data Hash, pk PublicKey, sig Signature) error {
 	return nil
 }
 
+// WriteSignedObject writes a length-prefixed object followed by its signature.
+func WriteSignedObject(w io.Writer, obj interface{}, sk SecretKey) error {
+	encObj := encoding.Marshal(obj)
+	sig, _ := SignHash(HashBytes(encObj), sk) // no error possible
+	return encoding.NewEncoder(w).EncodeAll(encObj, sig)
```

Comment:
Another thing that might make sense is to put the signature first. I think that's how nacl does it.